src/ai/backend/gateway/vfolder.py

```diff
@@ -910,6 +913,74 @@ def _do_ops():
     return web.json_response(resp, status=200)
 
 
+async def download_directory_as_archive(request: web.Request,
+                                        file_path: Path,
+                                        zip_filename: str = None) -> web.StreamResponse:
+    """Serve a directory as a zip archive on the fly."""
+    class _Sentinel:
+        pass
+
+    _sentinel = _Sentinel()
+
+    def _iter2aiter(iter):
+        """Iterable to async iterable"""
+        def _consume(loop, iter, q):
+            for item in iter:
+                while True:
+                    if not q.full():
+                        q.put(item)
+                        break
+                    time.sleep(0.01)
+            while True:
+                if not q.full():
+                    q.put(_sentinel)
+                    break
+                time.sleep(0.01)
```

Comment:
Since putting the sentinel object has negligible I/O and CPU overheads, I think we can just `q.put(_sentinel)` here.

Reply:
Reduce above code block with just `q.put(_sentinel)`.